### ISSUE TITLE
Crash after forced stop

### ISSUE THREAD

**Issue by simonpoole** [MarcusWolschon/osmeditor4android#2298]
Seems as if path pattern rendering crashes in Skia after a forced stop however this does not happen in simple examples.

Log up to crash looks normal with exception of lots of

```
06-01 16:37:32.254: D/TextView(22889): setTypeface with style : 0
06-01 16:37:32.254: D/TextView(22889): setTypeface with style : 0
06-01 16:37:32.254: D/TextView(22889): setTypeface with style : 0
06-01 16:37:32.254: D/TextView(22889): setTypeface with style : 0
06-01 16:37:32.264: D/TextView(22889): setTypeface with style : 0
```

Crash:

```
06-01 16:37:39.284: A/libc(22889): Fatal signal 6 (SIGABRT), code -6 in tid 22889 (de.blau.android)
06-01 16:37:39.414: A/DEBUG(328): *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
06-01 16:37:39.414: A/DEBUG(328): Build fingerprint: 'samsung/kltexx/klte:6.0.1/MMB29M/G900FXXS1CQD9:user/release-keys'
06-01 16:37:39.414: A/DEBUG(328): Revision: '14'
06-01 16:37:39.414: A/DEBUG(328): ABI: 'arm'
06-01 16:37:39.414: A/DEBUG(328): pid: 22889, tid: 22889, name: de.blau.android  >>> de.blau.android <<<
06-01 16:37:39.414: A/DEBUG(328): signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
06-01 16:37:39.444: A/DEBUG(328):     r0 00000000  r1 00005969  r2 00000006  r3 b6fc0b84
06-01 16:37:39.444: A/DEBUG(328):     r4 b6fc0b8c  r5 b6fc0b3c  r6 0000000c  r7 0000010c
06-01 16:37:39.444: A/DEBUG(328):     r8 00000008  r9 be9de80c  sl be9de608  fp 00000000
06-01 16:37:39.444: A/DEBUG(328):     ip 00000006  sp be9de570  lr b6d21c11  pc b6d24000  cpsr 400e0010
06-01 16:37:39.464: A/DEBUG(328): backtrace:
06-01 16:37:39.464: A/DEBUG(328):     #00 pc 00042000  /system/lib/libc.so (tgkill+12)
06-01 16:37:39.464: A/DEBUG(328):     #01 pc 0003fc0d  /system/lib/libc.so (pthread_kill+32)
06-01 16:37:39.464: A/DEBUG(328):     #02 pc 0001c393  /system/lib/libc.so (raise+10)
06-01 16:37:39.464: A/DEBUG(328):     #03 pc 00019611  /system/lib/libc.so (__libc_android_abort+34)
06-01 16:37:39.464: A/DEBUG(328):     #04 pc 00017564  /system/lib/libc.so (abort+4)
06-01 16:37:39.464: A/DEBUG(328):     #05 pc 0019567f  /system/lib/libskia.so (_Z16sk_realloc_throwPvj+12)
06-01 16:37:39.464: A/DEBUG(328):     #06 pc 00119fe1  /system/lib/libskia.so
06-01 16:37:39.464: A/DEBUG(328):     #07 pc 0011a7c5  /system/lib/libskia.so (_ZN9SkPathRef6EditorC1EP12SkAutoTUnrefIS_Eii+30)
06-01 16:37:39.464: A/DEBUG(328):     #08 pc 00118039  /system/lib/libskia.so (_ZN6SkPath7addPathERKS_RK8SkMatrixNS_11AddPathModeE+44)
06-01 16:37:39.464: A/DEBUG(328):     #09 pc 0015f687  /system/lib/libskia.so (_ZNK18SkPath1DPathEffect4nextEP6SkPathfR13SkPathMeasure+122)
06-01 16:37:39.464: A/DEBUG(328):     #10 pc 0015f51f  /system/lib/libskia.so (_ZNK14Sk1DPathEffect10filterPathEP6SkPathRKS0_P11SkStrokeRecPK6SkRect+74)
06-01 16:37:39.464: A/DEBUG(328):     #11 pc 00113c77  /system/lib/libskia.so (_ZNK7SkPaint11getFillPathERK6SkPathPS0_PK6SkRectf+56)
06-01 16:37:39.474: A/DEBUG(328):     #12 pc 00102c03  /system/lib/libskia.so (_ZNK6SkDraw8drawPathERK6SkPathRK7SkPaintPK8SkMatrixbbP9SkBlitter+358)
06-01 16:37:39.474: A/DEBUG(328):     #13 pc 000eb455  /system/lib/libskia.so (_ZN14SkBitmapDevice8drawPathERK6SkDrawRK6SkPathRK7SkPaintPK8SkMatrixb+24)
06-01 16:37:39.474: A/DEBUG(328):     #14 pc 000fba31  /system/lib/libskia.so (_ZN8SkCanvas10onDrawPathERK6SkPathRK7SkPaint+364)
06-01 16:37:39.474: A/DEBUG(328):     #15 pc 031d48bb  /system/framework/arm/boot.oat (offset 0x2f3b000)
06-01 16:37:40.364: W/ActivityManager(955): Launch timeout has expired, giving up wake lock!
06-01 16:37:40.364: A/DEBUG(328): Tombstone written to: /data/tombstones/tombstone_05
06-01 16:37:40.364: E/(328): ro.product_ship = true
06-01 16:37:40.364: E/(328): ro.debug_level = 0x4f4c
06-01 16:37:40.364: E/(328): sys.mobilecare.preload = false
```

**Comment by simonpoole** [comment]
I've tested back to (including) Android 9 and no problems back to that version. I suspect that force stopping leaves Skia (the hardware near Android graphics library) in a broken state in earlier versions of Android. Unluckily this means there isn't and likely never will be a real fix. 

I'll add a style without path patterns as workaround for users with older Android versions. Naturally not force stopping will avoid the issue.

**Comment by simonpoole** [comment]
19.0.2 adds a "Safe" shortcut that will set the data rendering style to the built-in minimal style (aka "black lines") and start the app. In my tests changing back to the regular style then works.

Note: shortcuts are only available on from Android 7.1 and later.
